src/memory_ops_common.jl

```diff
@@ -29,17 +29,46 @@ function add_vptr!(ls::LoopSet, array::Symbol, vptrarray::Symbol = vptr(array),
     end
     nothing
 end
-function subset_vptr!(ls::LoopSet, vptr::Symbol, indnum::Int, ind::Union{Symbol,Int})
+
+@inline valsum() = Val{0}()
+@inline valsum(::Val{M}) where {M} = Val{M}()
+@generated valsum(::Val{M}, ::Val{N}) where {M,N} = Val{M+N}()
+@inline valsum(::Val{M}, ::Val{N}, ::Val{K}, args...) where {M,N,K} = valsum(valsum(Val{M}(), Val{N}()), Val{K}(), args...)
+@inline valdims(::Any) = Val{1}()
+@inline valdims(::CartesianIndices{N}) where {N} = Val{N}()
+
+function append_loop_valdims!(valcall::Expr, loop::Loop)
+    if isstaticloop(loop)
+        push!(valcall.args, :(Val{1}()))
+    else
+        push!(valcall.args, Expr(:call, lv(:valdims), loop_boundary(loop)))
+    end
+    nothing
+end
+function subset_vptr!(ls::LoopSet, vptr::Symbol, indnum::Int, ind, previndices, loopindex)
```

Comment:
For now, I decided it would be easier to update the old `subsetview` function. This function is used for constant slices, like by `ifirst`. This function gets called very early, to avoid storing or representing that there was an index at all.
This probably caused the failures you observed, because it assumed the preceding indices corresponded to one axis each. That assumption was violated when `d == 1` and `d == 3`, where the first index corresponded to 0 and 2 axes, instead.

It now calls `valdims` on each of the preceding loops instead.
I currently defined `valdims` to return `Val(1)` for everything except `CartesianIndices{N}`, where it returns `Val(N)` instead.